### PR TITLE
[0.x] Adds support for in/excluded payment systems

### PR DESCRIPTION
**Pull request by lbebber**
#### What problem is this solving?

Currently, we support an argument in the `Installments` type called `installmentCriteria` which controls how many (and which) installment options should be returned in queries that ask for `Installments`.

However, some users need more advanced filter options, such as ignoring options from a certain `PaymentMethodName` or only taking into account options from a certain `PaymentMethodName`.

This PR adds support for two new arguments added to the `Installments` type by https://github.com/vtex-apps/search-graphql/pull/91.

#### How should this be manually tested?

Go to this [Workspace](https://victormiranda--storecomponents.myvtex.com/_v/private/vtex.search-graphql@0.33.0-beta.4/graphiql/v1?operationName=productSearchV3&variables=%7B%0A%20%20%22hideUnavailableItems%22%3A%20false%2C%0A%20%20%22skusFilter%22%3A%20%22ALL_AVAILABLE%22%2C%0A%20%20%22simulationBehavior%22%3A%20%22default%22%2C%0A%20%20%22installmentCriteria%22%3A%20%22MAX_WITHOUT_INTEREST%22%2C%0A%20%20%22productOriginVtex%22%3A%20false%2C%0A%20%20%22query%22%3A%20%22celulares-smartphones-e-smartwatches%2Fsmartphones%22%2C%0A%20%20%22orderBy%22%3A%20%22OrderByScoreDESC%22%2C%0A%20%20%22from%22%3A%200%2C%0A%20%20%22to%22%3A%2049%2C%0A%20%20%22excludedPaymentSystems%22%3A%20%5B%22Visa%22%2C%20%22Mastercard%22%5D%2C%0A%20%20%22selectedFacets%22%3A%20%5B%0A%20%20%20%20%7B%0A%20%20%20%20%20%20%22key%22%3A%20%22c%22%2C%0A%20%20%20%20%20%20%22value%22%3A%20%22celulares-smartphones-e-smartwatches%22%0A%20%20%20%20%7D%2C%0A%20%20%20%20%7B%0A%20%20%20%20%20%20%22key%22%3A%20%22c%22%2C%0A%20%20%20%20%20%20%22value%22%3A%20%22smartphones%22%0A%20%20%20%20%7D%0A%20%20%5D%0A%7D) and perform a query such as this stripped down `products` query taken from a category page:

```graphql
query productSearchV3($query: String, $excludedPaymentSystems: [String], $fullText: String, $selectedFacets: [SelectedFacetInput], $orderBy: String, $from: Int, $to: Int, $hideUnavailableItems: Boolean = false, $skusFilter: ItemsFilter = ALL_AVAILABLE, $simulationBehavior: SimulationBehavior = default, $installmentCriteria: InstallmentsCriteria = MAX_WITHOUT_INTEREST, $productOriginVtex: Boolean = false, $fuzzy: String, $operator: Operator, $searchState: String) {
  productSearch(query: $query, fullText: $fullText, selectedFacets: $selectedFacets, orderBy: $orderBy, from: $from, to: $to, hideUnavailableItems: $hideUnavailableItems, simulationBehavior: $simulationBehavior, productOriginVtex: $productOriginVtex, fuzzy: $fuzzy, operator: $operator, searchState: $searchState) {
    products {
      cacheId
      productId
      productName
      items(filter: $skusFilter) {
        name
        nameComplete
        sellers {
          sellerName
          commertialOffer {
            Installments(criteria: $installmentCriteria, excludedPaymentSystems: $excludedPaymentSystems) {
              InterestRate
              NumberOfInstallments
              Name
              PaymentSystemName
            }
          }
        }
      }
    }
  }
}
```

Tweak the values in the `excludedPaymentSystems` variable to see different results.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage

**Filtering out "Master Card" and "Visa" options**

Screen Shot 2020-09-14 at 16 00 37

**Filtering out "Visa" options**

Screen Shot 2020-09-14 at 16 01 00

**No filter**

Screen Shot 2020-09-14 at 16 01 17

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements 

#### Notes

Depends on: https://github.com/vtex-apps/search-graphql/pull/91

### CODE REVIEW

File: node/resolvers/search/offer.ts [hunk 1]
```diff
@@ -27,10 +43,25 @@ export const resolvers = {
...
+        filteredInstallments = filteredInstallments.filter(
+          ({ PaymentSystemName }) =>
+            includesCaseInsensitive(includedPaymentSystems, PaymentSystemName)
+        )
+      }
+
+      if (excludedPaymentSystems) {
+        filteredInstallments = filteredInstallments.filter(
+          ({ PaymentSystemName }) =>
+            (includesCaseInsensitive(excludedPaymentSystems, PaymentSystemName) === false)
```
Comment (lbebber) [inline node/resolvers/search/offer.ts:60]: `=== false` is more visible than a meager `!` at the start of the line

File: node/package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "ramda": "^0.26.1",
     "slugify": "^1.2.6",
     "typescript": "3.8.3",
-    "vtex.search-graphql": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.search-graphql@0.32.0/public"
+    "vtex.search-graphql": "http://vtex.vtexassets.com/_v/public/typings/v1/vtex.search-graphql@0.33.0-beta.4/public"
```
Comment (victorhmp) [inline node/package.json:34]: Reminder to update this once https://github.com/vtex-apps/search-graphql/pull/91 is merged.

**Review by lbebber — COMMENTED** [review]
LGTM :spidermen:

**Review by victorhmp — APPROVED** [review]

**Review by thalytafabrine — APPROVED** [review]

File: node/resolvers/search/offer.ts (hunk 1 above)
Comment (klzns) [reply node/resolvers/search/offer.ts:60]: Why filter by name, is the ID not available?

**Review by klzns — CHANGES_REQUESTED** [review]

File: node/resolvers/search/offer.ts (hunk 1 above)
Comment (victorhmp) [reply node/resolvers/search/offer.ts:60]: Yep, the ID is not available in this type.

Also, `PaymentSystemName` seems more intuitive to me. The use case we based this on came from a request like "We want to fetch the installment option with the maximum number of payments, no interest **and that is not Cartão Carrefour**"

**Review by klzns — APPROVED** [review]

**Pull request merged**